plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,196 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/bluenviron/gomavlib/v3"
+	"github.com/bluenviron/gomavlib/v3/pkg/dialects/ardupilotmega"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Mavlink struct {
+	URL                    string   `toml:"url"`
+	SystemID               uint8    `toml:"system_id"`
+	FilterPattern          []string `toml:"filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	filter         filter.Filter
+	connection     *gomavlib.Node
+	endpointConfig []gomavlib.EndpointConf
+	cancel         context.CancelFunc
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Mavlink) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Mavlink) Init() error {
+	// Parse out the Mavlink endpoint.
+	// Try to parse the URL
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return fmt.Errorf("invalid url: %w", err)
+	}
+
+	// Split host and port, and use default port if it was not specified
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		// Use default port if we could not parse out the port.
+		host = u.Host
+		port = "14550"
+	}
+
+	switch u.Scheme {
+	case "serial":
+		// Serial client
+		// Parse serial URL by hand, because it is not a compliant URL.
+		baudRate := 57600
+		device, rate, found := strings.Cut(strings.TrimPrefix(s.URL, "serial://"), ":")
+		if found {
+			r, err := strconv.Atoi(rate)
+			if err != nil {
+				return fmt.Errorf("serial baud rate not valid: %w", err)
+			}
+			baudRate = r
+		}
+
+		s.endpointConfig = []gomavlib.EndpointConf{
+			gomavlib.EndpointSerial{
+				Device: device,
+				Baud:   baudRate,
+			},
+		}
+	case "tcp":
+		if len(host) > 0 {
+			s.endpointConfig = []gomavlib.EndpointConf{
+				gomavlib.EndpointTCPClient{
+					Address: host + ":" + port,
+				},
+			}
+		} else {
+			s.endpointConfig = []gomavlib.EndpointConf{
+				gomavlib.EndpointTCPServer{
+					Address: ":" + port,
+				},
+			}
+		}
```

Comment:
This can be simplified to
```suggestion
		s.endpointConfig = []gomavlib.EndpointConf{
			gomavlib.EndpointTCPClient{
				Address: host + ":" + port,
			},
		}
```
same for UDP.

Reply:
In this case it can't be simplified because the internal implementation of `EndpointTCPClient` and `EndpointTCPServer` are different (same for UDP) so although the parameters are the same the types differ.